cmd/earthly/main.go

```diff
@@ -1316,12 +1346,19 @@ func (app *earthlyApp) configureSatellite(cc cloud.Client) error {
 	return nil
 }
 
-func (app *earthlyApp) isUsingSatellite() bool {
-	return len(app.cfg.Satellite.Name) > 0
+func (app *earthlyApp) isUsingSatellite(c *cli.Context) bool {
+	if app.noSatellite {
+		return false
+	}
+	if c.IsSet("buildkit-host") {
+		// buildkit-host takes precedence
+		return false
+	}
+	return len(app.cfg.Satellite.Name) > 0 || app.satelliteName != ""
```

Comment:
Shouldn't the signal here be both org and name, now that we've extended it into that tuple to specify a satellite name?

Reply:
Org isn't strictly necessary - for example, if a user is only part of one org they can specify only the satellite name and we'll default to that org automatically.

Reply:
It is? Shouldnt that be accounted for here if org is implicit like that? What if they use `--sat` with no org or login?

Reply:
nit: check for empty-string the same way on both conditions?

```
return app.cfg.Satellite.Name != ""  || app.satelliteName != ""
```

Reply:
So this is the error shown when an org is not specified and cannot be deduced:

<img width="1044" alt="Screen Shot 2022-06-28 at 12 00 44 PM" src="https://user-images.githubusercontent.com/446771/176261653-50007fb0-47fd-4213-bc90-c282e6a7f55b.png">

The implicitness happens in `getSatelliteOrgID`, which is called a little lower in the same function.

With this implementation, it is allowed to skip specifying the org in the following two cases:

1. User is part of only a single org
2. A satellite is selected, in which case we use the same org of the selected satellite